### PR TITLE
Update cli with `yarn tarball` for binary generation

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Updates dependency of CLI package on @celo/dev-cli with new upstream features

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "docs": "yarn oclif-dev readme --multi --dir=../docs/command-line-interface && yarn prettier ../docs/command-line-interface/*.md --write",
     "lint": "tslint -c tslint.json --project tsconfig.json",
     "prepack": "yarn run build && oclif-dev manifest && oclif-dev readme",
+    "tarball:": "npm shrinkwrap && yarn oclif-dev pack && rm npm-shrinkwrap.json",
```
Comment (mcortesi) [inline packages/cli/package.json:29]: i had a really bad time with `npm shrinkwrap` in the past... and then came `yarn`.... so i'm a bit hesistant to use it here. Also... take in consideration that the moment you use `npm`; it won't read `yarn.lock` so you're building something different

**Comment by barbaraliau** [comment]
@yorhodes is this PR still relevant?

**Comment by yorhodes** [comment]
> @yorhodes is this PR still relevant?

I think it's useful if we want a standalone binary such that users don't need specific node/npm versions/installations
https://github.com/oclif/dev-cli#oclif-dev-pack

This is also relevant https://github.com/celo-org/celo-monorepo/issues/3661

**Comment by nambrot** [comment]
Closing for now, feel free to re-open if still applicable

**Pull request closed without merging**